src/object/object_primitive.c

```diff
@@ -17145,166 +17162,157 @@ static int
 mr_getmem_json (void *memptr, TP_DOMAIN * domain, DB_VALUE * value, bool copy)
 {
   int error = NO_ERROR;
-  DB_JSON *json_obj = STATIC_CAST (DB_JSON *, memptr);
+  char *json_raw_body = NULL;
+  const char *json_schema = NULL;
+  DB_JSON *json, json_copy;
+  JSON_DOC *new_doc = NULL;
+
+  json = (DB_JSON *) memptr;
 
-  if (json_obj == NULL || json_obj->json_body == NULL)
+  if (json == NULL)
     {
+      DB_MAKE_NULL (value);
       db_value_domain_init (value, DB_TYPE_JSON, domain->precision, 0);
       value->need_clear = false;
       return NO_ERROR;
     }
-
-  if (!copy)
-    {
-      db_make_json (value, json_obj->json_body, json_obj->document, false);
-    }
   else
     {
-      char *new_ = NULL;
-      JSON_DOC *document = NULL;
-
-      new_ = db_private_strdup (NULL, json_obj->json_body);
-      if (new_ == NULL)
+      if (!copy)
 	{
-	  assert (er_errid () != NO_ERROR);
-	  error = er_errid ();
+	  json_raw_body = json->json_body;
```

Comment:
when copy is false, are you sure destination json should be cleared? I suspect it should not.